test/presubmit-tests.sh

```diff
@@ -43,7 +46,21 @@ function unit_tests() {
 
 function integration_tests() {
   echo "Running integration tests"
-  echo "TODO(#16): add integration tests"
+
```

Comment:
> I strongly suggest that you move this to a separate script (and even call it e2e-tests.sh).
Source the e2e-tests.sh in vendor. You don't need a separate cluster-setup.sh file, it's just 2 helpers. 1 script is enough.

i'm kind of confused - right now I have `presubmit-tests.sh` and `cluster-setup.sh`, which i think does what you're suggesting `e2e-tests.sh` does, except it doesn't actually call `report_go_test` - is it important that `report_go_test` is called from the other script?

> You're not emitting metrics, so you can omit that part for now, for simplicity.

I think since I've included all the machinery in the tests to be able to emit metrics I might as well keep it

Reply:
> is it important that report_go_test is called from the other script?

No. But `e2e-tests.sh` assumes it's being called from the main script, so it can call kubetest and hide the complexity from the main script. Since you're sourcing it from inside a function, there's a good chance of conflicts (e.g., flags) and weird flows.

If your idea is that `cluster-setup.sh` is a helper script to setup an existing cluster for your tests, you don't need (and actually shouldn't) source `e2e-tests.sh`. Just assume the environment is ready and do your stuff.

> You're not emitting metrics, so you can omit that part for now, for simplicity.

Outdated comment, now that the helper function is available. You get metrics for free, without cluttering your script.